src/main/java/org/opensearch/knn/index/util/IndexUtil.java

```diff
@@ -170,20 +162,18 @@ public static ValidationException validateKnnField(
                 return exception;
             }
 
-            // Block binary vector data type for pq encoder
+            // Block binary and byte vector data type for any encoder
             if (trainRequestKnnMethodContext != null) {
                 MethodComponentContext methodComponentContext = trainRequestKnnMethodContext.getMethodComponentContext();
                 Map<String, Object> parameters = methodComponentContext.getParameters();
 
                 if (parameters != null && parameters.containsKey(KNNConstants.METHOD_ENCODER_PARAMETER)) {
                     MethodComponentContext encoder = (MethodComponentContext) parameters.get(KNNConstants.METHOD_ENCODER_PARAMETER);
-                    if (encoder != null
-                        && KNNConstants.ENCODER_PQ.equals(encoder.getName())
-                        && VectorDataType.BINARY == trainRequestVectorDataType) {
+                    if (encoder != null && VECTOR_DATA_TYPES_NOT_SUPPORTING_ENCODERS.contains(trainRequestVectorDataType)) {
```

Comment:
Thanks for the update. Btw, could you also check if the right datatype is validated when training?

Reply:
It is already taken care 
https://github.com/opensearch-project/k-NN/pull/2002/files#diff-572fd1fb31f6ec72d21051e6c123948cf758179a61fc6e50349f0be8123bd553R41